internal/util/importutilv2/csv/row_parser.go

```diff
@@ -115,22 +140,47 @@ func (r *rowParser) Parse(strArr []string) (Row, error) {
 		} else if r.dynamicField != nil {
 			dynamicValues[r.header[index]] = value
 		} else {
-			return nil, fmt.Errorf("the field '%s' is not defined in schema", r.header[index])
+			// from v2.6, we don't intend to return error for redundant fields, just skip it
+			continue
+			// return nil, merr.WrapErrImportFailed(
+			// 	fmt.Sprintf("the field '%s' is not defined in schema", r.header[index]))
+		}
+	}
+
+	// if nullable/defaultValue fields have no values, fill with nil or default value
```

Comment:
nullable/defaultValue fields will be filled in `internal/datanode/importv2/task_import.go`, why we need to fill nullable/default here?

Reply:
The rowParser.Parse() returns a Row object, Row = map[storage.FieldID]any
In csv.reader.Read(), the row object is passed to insertData.Append()
row, err := r.parser.Parse(value)
err = insertData.Append(row)

The InsertData.Append() calls FieldData.AppendRow() for each field
In csv/json, for nullable fields, some values could be not-null, some values could be null. If we don't fill the Row with null/default values, the row number of nullable fields could be different to other fields.